lib/memery.rb

```diff
@@ -6,16 +6,34 @@
 
 module Memery
   class << self
-    def included(base)
-      base.extend(ClassMethods)
-      base.include(InstanceMethods)
-    end
-
     def monotonic_clock
       Process.clock_gettime(Process::CLOCK_MONOTONIC)
     end
   end
 
+  OUR_BLOCK = lambda do
+    extend(ClassMethods)
+    include(InstanceMethods)
+    extend ModuleMethods if instance_of?(Module)
+  end
+
+  private_constant :OUR_BLOCK
+
+  module ModuleMethods
+    def included(base = nil, &block)
```

Comment:
This looks like a hack :(
What if `self.included` method has some other signature?

Reply:
> This looks like a hack :(

ActveSupport has hacks itself, sadly: https://github.com/rails/rails/blob/0a4a4f9d84caa7e39915eb641a8c1e5c4fb426b8/activesupport/lib/active_support/concern.rb#L156-L168

> What if self.included method has some other signature?

Please, describe it more clear. I thought that `self.included` always must to have 1 argument, and then (because of your changes) I've met ActiveSupport code/logic/usage/hacks/monkey-patches, and I've adopt our (Memery) code to ActiveSupport. I think it's a good solution for your task. If we will face with other `self.include` implementations and hacks/monkey-patches — we can adopt our code for them too (or it can be done locally, as you know, thanks to Ruby).